### DISCUSSION TITLE
Can the project expose an API endpoint so external applications can make requests?

### DISCUSSION THREAD

**Discussion by beaver316** [acheong08/ChatGPT#337] [General]
I'm looking to create a proof of concept where I create my own front end application which makes requests to the ChatGPT. From what I've seen with this project you only interact with OpenAI's API through the terminal of Python. Does this project have support for exposing an API where I can POST my request and receive the answer as a JSON? If not, is this something planned?

I found a project on GitHub doing exactly this, but the project has not been updated since OpenAI implemented Cloudfare.

https://github.com/chitalian/revChatGPTServer

**Comment by ausbitbank** [comment]
See the commented out api code at the top of this file. Sounds like you just need a simple python script that uses the library + flask to expose an api

https://github.com/ausbitbank/ChatGPTDiscord/blob/main/ChatGPTdiscord.py
